ob1k-cache/src/main/java/com/outbrain/ob1k/cache/LocalAsyncCache.java

```diff
@@ -112,9 +112,22 @@ public LocalAsyncCache() {
 
   private ComposableFuture<V> loadElement(final K key) {
     final ComposableFuture<V> loadedElement = loader.load(cacheName, key).materialize();
-    return loadedElement.continueOnError((FutureErrorHandler<V>) error -> {
-      loadingCache.asMap().remove(key, loadedElement);
-      return fromError(error);
+    return loadedElement;
+  }
+
+  private ComposableFuture<V> extractMapEntry(ComposableFuture<Map<K, V>> loaded ,K key) {
```

Comment:
and last, but this is just mine opinion: you shouldn't accept the future as a param, rather just return the FutureSuccessHandler, and do the continuation outside.